index.d.ts

```diff
@@ -5,33 +5,46 @@ interface ClearablePromise<T> extends Promise<T> {
 	clear(): void;
 }
 
+interface DelayOptions {
+	/**
+	 * An optional AbortSignal to abort the delay.
+	 * If aborted, the Promise will be rejected with an AbortError.
+	 */
+	signal?: AbortSignal
+}
+
 declare const delay: {
 	/**
 	 * Create a promise which resolves after the specified `milliseconds`.
 	 *
 	 * @param milliseconds - Milliseconds to delay the promise.
 	 * @returns A promise which resolves after the specified `milliseconds`.
 	 */
-	(milliseconds: number): ClearablePromise<void>;
+	(milliseconds: number, options?: DelayOptions): ClearablePromise<void>;
 
 	/**
 	 * Create a promise which resolves after the specified `milliseconds`.
 	 *
 	 * @param milliseconds - Milliseconds to delay the promise.
-	 * @param value - Value to resolve in the returned promise.
 	 * @returns A promise which resolves after the specified `milliseconds`.
 	 */
-	<T>(milliseconds: number, value: T): ClearablePromise<T>;
+	<T>(milliseconds: number, options?: DelayOptions & {
+		/** Value to resolve in the returned promise. */
+		value: T
```

Comment:
Why is this not made part of `DelayOptions`?

Reply:
If it was (and of type `T`) you could not pass only `signal`, you would always also have to pass `{ value: undefined }`. `value` could be made optional with `?` but then you could do `const x: number = delay<number>(100, { value: undefined })`, which would be wrong, cause it resolves with `undefined`, not `number`.

This way there is an overload that specifically handles no `value` being passed and resolving `void`, and one where `value` is passed with type `T` resolving `T`.